04/app.js

```diff
@@ -1,8 +1,37 @@
-console.log('DOM');
+const nav = document.querySelector('nav');
+const ul = document.createElement('ul');
+const liList = [];
+const aList = [];
+
+for(let i = 0; i<3; i++) {
+    liList[i] = document.createElement('li');
+    aList[i] = document.createElement('a')
+    ul.appendChild(liList[i]);
+    liList[i].appendChild(aList[i]);
+}
+nav.appendChild(ul);
+
+
+
+
 
 // struktura do wykorzystania w pętli
 const menuItems = [
     {text: 'start', url: '/'},
     {text: 'galeria', url: '/gallery'},
     {text: 'kontakt', url: '/contact'},
-];
\ No newline at end of file
+];
+
+aList.forEach(function(item, i) {
+    item.setAttribute('href', menuItems[i].url);
+    item.innerText = menuItems[i].text;
+})
```

Comment:
Może tą część dałoby się od razu zrobić w pętla `for`? (pamiętaj, aby przenieść zmienną `menuItems` przed pętlą).
```
for(let i = 0; i<3; i++) {
    liList[i] = document.createElement('li');
    aList[i] = document.createElement('a')
    ul.appendChild(liList[i]);
    liList[i].appendChild(aList[i]);
    
    aList[i].setAttribute('href', menuItems[i].url); // pobranie wartości z tablicy, która zawiera obiekt
}
```

Ten kod powyżej też powinien posiadać parę usprawnień. 
Niepotrzebne tworzysz dodatkowe struktury danych (tablice) oraz nie wykorzystujesz informacji o ilości elementó w tablicy, lepiej by było:
```
for(let i = 0; i<menuItems.lenght; i++) { // zmiana w menuItems nie spowoduje blednego działania
    const li = document.createElement('li'); // nie ma potrzeby wykorzystywać tablicy
    const a = document.createElement('a') // potrzebuejmy te zmienne tyko na chwile ;)
    ul.appendChild(li);
    li.appendChild(a);
    
    a.setAttribute('href', menuItems[i].url); // pobranie wartości z tablicy, która zawiera obiekt
}
```